specs/phase1/data-availability-sampling.md

```diff
@@ -0,0 +1,218 @@
+# Ethereum 2.0 Phase 1 -- Data Availability Sampling
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Custom types](#custom-types)
+- [Configuration](#configuration)
+  - [Misc](#misc)
+- [New containers](#new-containers)
+  - [DASSample](#dassample)
+  - [ShardBlob](#shardblob)
+  - [SignedShardBlob](#signedshardblob)
+- [Helper functions](#helper-functions)
+  - [Reverse bit ordering](#reverse-bit-ordering)
+    - [`reverse_bit_order`](#reverse_bit_order)
+    - [`reverse_bit_order_list`](#reverse_bit_order_list)
+  - [Data extension](#data-extension)
+  - [Data recovery](#data-recovery)
+- [DAS functions](#das-functions)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Custom types
+
+We define the following Python custom types for type hinting and readability:
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SampleIndex` | `uint64` | A sample index, corresponding to chunk of extended data |
+| `BLSPoint` | `uint256` | A number `x` in the range `0 <= x < MODULUS` |
```

Comment:
Suggest `FieldElement` instead?

Reply:
For consistency with the other BLS related types, I kind of liked the prefix in the type, and then just went with the shortest name. FieldElement is good too, maybe better, no strong opinion here.

Reply:
Also, we already have `POINTS_PER_SAMPLE`. So I will keep this as-is for consistency, but we can revisit it later, and do a full sweep of typing/naming improvements.

Reply:
I like FieldElement because it would make it clear that this is a much simpler object than a BLS curve point. The only thing it has to do with BLS is really that it happens to use the BLS curve order as modulus.